src/DataGrid.tsx

```diff
@@ -1151,11 +1114,12 @@ function DataGrid<R, SR, K extends Key>(
           ...style,
           // set scrollPadding to correctly position non-sticky cells after scrolling
           scrollPaddingInlineStart:
-            selectedPosition.idx > lastFrozenColumnIndex
+            selectedPosition.idx > lastFrozenColumnIndex || scrollToPosition?.idx !== undefined
```

Comment:
Why do we need this?

Reply:
This is needed otherwise cell will be behind the frozen columns or header/summary rows. T

https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-padding-inline-start
> The scroll-padding-inline-start property defines offsets for the start edge in the inline dimension of the optimal viewing region of the scrollport: the region used as the target region for placing things in view of the user.

Reply:
Ah right I misunderstood.
Now my question is: when do we want `scrollPaddingInlineStart` to be `undefined`? Would it simplify things if it was always set? Even if it's `0px`

Reply:
It seems like it is needed
![Recording 2023-05-25 at 11 57 52](https://github.com/adazzle/react-data-grid/assets/1452717/3fd3812e-ec45-43c1-a125-44c9e4872931)

Reply:
Might be an issue in Chrome, I don't think I can reproduce the same issue in Firefox.
Should open a chromium issue if there isn't one already.